src/amo/reducers/reviews.js

```diff
@@ -72,70 +74,48 @@ export type ReviewsState = {|
   byAddon: ReviewsByAddon,
   byId: ReviewsById,
   byUserId: ReviewsByUserId,
+  latestByAddonVersion: {
+    // The latest user review for add-on / version
+    // or null if one does not exist yet.
+    [userIdAddonIdVersionId: string]: number | null,
+  },
   view: {
     [reviewId: number]: ViewStateByReviewId,
   },
-
-  // This is what the current data structure looks like:
-  // [userId: string]: {
-  //   [addonId: string]: {
-  //     [reviewId: string]: UserReviewType,
-  //   },
-  // },
-  //
-  // TODO: make this consistent by moving it from state[userId] to
-  // state.byUser[userId]
-  // https://github.com/mozilla/addons-frontend/issues/1791
-  //
-  // Also note that this needs to move to state.byUser before its type
-  // can be expressed in Flow without conflicting with state.byAddon.
-  //
 |};
 
 export const initialState: ReviewsState = {
   byAddon: {},
   byId: {},
   byUserId: {},
+  latestByAddonVersion: {},
   // This stores review-related UI state.
   view: {},
 };
 
-type ExpandReviewObjectsParams = {|
-  state: ReviewsState,
-  reviews: Array<number>,
-|};
+export const selectReview = (
+  reviewsState: ReviewsState,
+  id: number,
+): UserReviewType | void => {
```

Comment:
Grr. I thought that `void` was a way to express "this function does not return anything" but Flow actually uses `void` to express `undefined`... 🤦‍♂️

Reply:
Isn't that the same, though? Not returning anything is the same as returning a literal `undefined`.